beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -30,24 +32,38 @@ enum RangeBlockDataRequest<E: EthSpec> {
             DataColumnsByRangeRequestId,
             ByRangeRequest<DataColumnsByRangeRequestId, DataColumnSidecarList<E>>,
         >,
+        /// The column indices corresponding to the request
+        column_peers: HashMap<DataColumnsByRangeRequestId, Vec<ColumnIndex>>,
         expected_custody_columns: Vec<ColumnIndex>,
     },
 }
 
+#[derive(Debug)]
+pub struct CouplingError {
+    pub(crate) msg: String,
+    pub(crate) column_and_peer: Option<(Vec<(ColumnIndex, PeerId)>, PeerAction)>,
+}
```

Comment:
You could make this an enum

```rs
enum CouplingError {
    InternalError(String),
    CustodyFailure(String, Vec<(u64, PeerId)>, PeerAction),
}
```

Reply:
I like this, will do it in the backfill PR